pkg/app/pipedv1/livestatereporter/livestatereporter.go

```diff
@@ -45,127 +51,212 @@ type Reporter interface {
-	pr.logger.Info("start running app live state reporter", zap.Duration("snapshot-flush-interval", pr.snapshotFlushInterval))
+func (r *reporter) Run(ctx context.Context) error {
+	r.logger.Info("start running app live state reporter", zap.Duration("snapshot-flush-interval", r.snapshotFlushInterval))
 
-	snapshotTicker := time.NewTicker(pr.snapshotFlushInterval)
+	snapshotTicker := time.NewTicker(r.snapshotFlushInterval)
 	defer snapshotTicker.Stop()
 
 	for {
 		select {
 		case <-snapshotTicker.C:
-			pr.flushSnapshots(ctx)
+			r.flushSnapshots(ctx)
 
 		case <-ctx.Done():
-			pr.logger.Info("app live state reporter has been stopped")
+			r.logger.Info("app live state reporter has been stopped")
 			return nil
 		}
 	}
 }
 
-func (pr *pluginReporter) flushSnapshots(ctx context.Context) {
-	// TODO: Implement appLister.ListByPluginName.
-	apps := pr.appLister.ListByPluginName(pr.pluginName)
-	for _, app := range apps {
-		res, err := pr.pluginClient.GetLivestate(ctx, &livestate.GetLivestateRequest{ApplicationId: app.Id})
+func (r *reporter) flushSnapshots(ctx context.Context) {
+	apps := r.appLister.List()
+
+	repoMap := make(map[string]git.Repo)
+	for id, cfgRepo := range r.pipedConfig.GetRepositoryMap() {
+		repo, err := r.gitClient.Clone(ctx, id, cfgRepo.Remote, cfgRepo.Branch, fmt.Sprintf("%s/%s", r.workingDir, id))
 		if err != nil {
-			pr.logger.Info(fmt.Sprintf("no app state of application %s to report", app.Id))
+			r.logger.Error("failed to clone repository", zap.String("repo-id", id), zap.Error(err))
 			continue
 		}
+		defer repo.Clean()
+		repoMap[id] = repo
+	}
+
+	// TODO: Set the limit based on the piped config
+	limit := runtime.GOMAXPROCS(0) / 2
 
-		// Report the application live state to the control plane.
-		snapshot := &model.ApplicationLiveStateSnapshot{
-			ApplicationId:        app.Id,
-			PipedId:              app.PipedId,
-			ProjectId:            app.ProjectId,
-			Kind:                 app.Kind,
-			ApplicationLiveState: res.GetApplicationLiveState(),
+	appsPerReporter := len(apps) / limit
+	if appsPerReporter == 0 {
+		appsPerReporter = 1
+	}
+
+	appsGrouped := make([][]*model.Application, 0)
+	for i := 0; i < len(apps); i += appsPerReporter {
+		end := i + appsPerReporter
+		if end > len(apps) {
+			end = len(apps)
 		}
-		snapshot.DetermineApplicationHealthStatus()
-
-		if _, err := pr.apiClient.ReportApplicationLiveState(ctx, &pipedservice.ReportApplicationLiveStateRequest{
-			Snapshot: snapshot,
-		}); err != nil {
-			pr.logger.Error("failed to report application live state",
-				zap.String("application-id", app.Id),
-				zap.Error(err),
-			)
+		appsGrouped = append(appsGrouped, apps[i:end])
+	}
+
+	eg, ctx := errgroup.WithContext(ctx)
+	eg.SetLimit(limit)
+
+	r.logger.Info("start flushing snapshots", zap.Int("application-count", len(apps)), zap.Int("concurrency", limit))
+	for _, apps := range appsGrouped {
+		eg.Go(func() error {
+			r.flushAll(ctx, apps, repoMap)
+			return nil
+		})
+	}
```

Comment:
At first, I tried to implement the livestate reporter to process the apps grouped by plugins because the livestatereporter for pipedv0 processes the apps grouped by platform providers.

But it finds the plugin name of the app based on the app config. 
It takes some more time and resource to prepare the deploy sources for all of the apps in one time.
So I implemented it in paralell.